### ISSUE TITLE
[Question]: Is non-provider-responsible-role-references-user supposed to link to "privilege-uuid" in the root SSP?  

### ISSUE THREAD

**Issue by JoseGHdz** [GSA/fedramp-automation#977]
### This is a ...

question - need to understand something

### This relates to ...

- [ ] the **FedRAMP OSCAL Registry**
- [ ] the **FedRAMP OSCAL baselines**
- [ ] the **Guide to OSCAL-based FedRAMP Content**
- [X] the **Guide to OSCAL-based FedRAMP System Security Plans (SSP)**
- [ ] the **Guide to OSCAL-based FedRAMP Security Assessment Plans (SAP)**
- [ ] the **Guide to OSCAL-based FedRAMP Security Assessment Results (SAR)**
- [ ] the **Guide to OSCAL-based FedRAMP Plan of Action and Milestones (POA&M)**
- [X] the **FedRAMP SSP OSCAL Template** (JSON or XML Format)
- [ ] the **FedRAMP SAP OSCAL Template** (JSON or XML Format)
- [ ] the **FedRAMP SAR OSCAL Template** (JSON or XML Format)
- [ ] the **FedRAMP POA&M OSCAL Template** (JSON or XML Format)
- [X] the **FedRAMP OSCAL Validations**

## **What's your feedback?**  
I am encountering an issue with the **"non-provider-responsible-role-references-user"** constraint validation. The error message states:  
> _"A FedRAMP SSP MUST have each component describing leveraged systems, interconnections, or authorized services reference at least one user with an authorized privilege and function performed via the 'privilege-uuid' property."_  

However, the validation system points to the overall SSP location:  
**`/system-security-plan`**  

This behavior creates ambiguity because the problem originates from specific components that lack the `privilege-uuid` property within their `responsible-role` definitions. Checking this constraint at the **component level** under:  
**`/system-security-plan/system-implementation/components`**  
would provide more actionable and precise feedback.

---

## **Where Exactly?**  
This is related to the constraint:  
**`non-provider-responsible-role-references-user`**  

The issue arises because the validation error points to `/system-security-plan` instead of the specific components that violate the rule:  
**`/system-security-plan/system-implementation/components`**  

### Supporting Information:
- **Constraint:** `non-provider-responsible-role-references-user`  
- **Documentation Reference:** [External Systems and Services Not Having FedRAMP Authorization](https://automate.fedramp.gov/documentation/ssp/4-ssp-template-to-oscal-mapping/#external-systems-and-services-not-having-fedramp-authorization)

---

### **Expected Behavior:**  
The error should be reported at the **component level** where the issue originates. This approach provides a clear and actionable location to address the missing `privilege-uuid`.

### **Actual Behavior:**  
The validation system reports the error at the SSP root: `/system-security-plan`, making it challenging to identify which component is responsible.

---

### **Request for Assistance:**  
Could you confirm if this constraint is intended to operate at the **component level**? If so, can the validation system be updated to point to the specific path where the error occurs?  

Any clarification or further guidance would be greatly appreciated.

**Comment by aj-stein-gsa** [comment]
Hi @JoseGHdz, thanks so much for this report. I have not closed out the related development task and the documentation work is potentially still pending. We got ahead of ourselves, apologies. I will update the documentation in a PR before release and also ask you for your review.

In the interim, you may want to review a valid example with more detail and inline remarks that you may find informative.

https://github.com/GSA/fedramp-automation/blob/99f532f2828eb9782612527b64326732fcc1f5cc/src/content/rev5/examples/ssp/xml/fedramp-ssp-example.oscal.xml#L1492-L1567

**Comment by aj-stein-gsa** [comment]
Just a heads up, @JoseGHdz, since you posed this as a question, I will only mark it blocked until I hear back from you as the author to ensure this helps. Since the answer involves completing and publishing the documentation, not the example, this question issue will not get resolved until those tasks are achieved. I will just want to check back with you on the interim and confirm this sample helps you. Either way, let us know.

**Comment by JoseGHdz** [comment]
> Just a heads up, @JoseGHdz, since you posed this as a question, I will only mark it blocked until I hear back from you as the author to ensure this helps. Since the answer involves completing and publishing the documentation, not the example, this question issue will not get resolved until those tasks are achieved. I will just want to check back with you on the interim and confirm this sample helps you. Either way, let us know.

This looks great! I just went through the sample documentation, and based on this example:  
[https://github.com/GSA/fedramp-automation/blob/99f532f2828eb9782612527b64326732fcc1f5cc/src/content/rev5/examples/ssp/xml/fedramp-ssp-example.oscal.xml#L1023-L1078](https://github.com/GSA/fedramp-automation/blob/99f532f2828eb9782612527b64326732fcc1f5cc/src/content/rev5/examples/ssp/xml/fedramp-ssp-example.oscal.xml#L1023-L1078)  

It’s pretty similar to what I have, and I can see the changes I need to make.  

**One thing I noticed**: The validation error is currently pointing to the root level, which makes it harder to track down the exact component that needs fixing for the "privilege-uuid".  

Would it be possible to have the error reported at the **/system-security-plan/system-implementation/components** level instead of **/system-security-plan**? That would make it much easier to pinpoint where the issue is with the validation tool.

**Comment by aj-stein-gsa** [comment]
We appreciate the feedback that is a sensible request. I will follow up later today or tomorrow with a comment about the decision and status updates if we make a code change. I'd say that it is quite likely we'll do that, @JoseGHdz. 😁

**Comment by JoseGHdz** [comment]
> We appreciate the feedback that is a sensible request. I will follow up later today or tomorrow with a comment about the decision and status updates if we make a code change. I'd say that it is quite likely we'll do that, @JoseGHdz. 😁

Awesome! Thanks for all the help. I really appreciate it.

**Comment by aj-stein-gsa** [comment]
> Would it be possible to have the error reported at the **/system-security-plan/system-implementation/components** level instead of **/system-security-plan**? That would make it much easier to pinpoint where the issue is with the validation tool.


I just wanted to follow up after the last few days of issues. I thought your follow-on request was for path retargeting was in #999, but it seems not. You still want this change, but I should open it up in a separate issue, correct? I just wanted to make sure I understood constraint changes and documentation touch-ups we do based upon the issue original reported here, @JoseGHdz. Thanks!

**Comment by JoseGHdz** [comment]
> > Would it be possible to have the error reported at the **/system-security-plan/system-implementation/components** level instead of **/system-security-plan**? That would make it much easier to pinpoint where the issue is with the validation tool.
> 
> I just wanted to follow up after the last few days of issues. I thought your follow-on request was for path retargeting was in #999, but it seems not. You still want this change, but I should open it up in a separate issue, correct? I just wanted to make sure I understood constraint changes and documentation touch-ups we do based upon the issue original reported here, @JoseGHdz. Thanks!

I wanted to clarify the difference between issue #999 and issue #977.

Issue #999 was about a specific error I ran into, whereas issue #977 is more focused on how the validation system reports those errors.

After digging deeper, I realized the problem with #999 was actually on my end. I missed a key detail in the documentation that Leveraged Authorization linking can only be applied to items of type "system", not "service". Since I was using AWS and Azure services in the components section, I had set their type to "service", which caused the error. That’s why I ended up closing #999.

However, this ties directly back to issue #977. The validation system didn’t give me enough information to figure out where the problem was or how to fix it. Instead, the error was just pointed at the root item, which made it harder to identify the actual issue.

I hope this helps clarify things. It also highlights how improving the validation system’s feedback would make it a lot easier to track down and fix these kinds of errors.
